heartbeat/monitors/monitor.go

```diff
@@ -175,6 +182,9 @@ func newMonitorUnsafe(
 
 		logp.L().Error(fullErr)
 		p.Jobs = []jobs.Job{func(event *beat.Event) ([]jobs.Job, error) {
+			// if statusReporter is set, as it is for running managed-mode, update the input status
+			// to failed, specifying the error
+			m.updateStatus(status.Failed, fmt.Sprintf("monitor could not be started: %s, err: %s", m.stdFields.ID, fullErr))
```

Comment:
Reading this 
```
	// Failed is status describing unit is failed. This status should
	// only be used in the case the beat should stop running as the failure
	// cannot be recovered.
```
Could this cause other HB to stop and also other monitors from running? Is this intended?

Reply:
> Could this cause other HB to stop and also other monitors from running?

It probably won't (it doesn't, as of now). Even if that were the case, since the status is scoped at monitor level, it should only filter the failed integrations, but I'm speculating here. There are also multiple status layers, this change only affects the stream (not even the integration) status.
As for the status, either `failed` or `degraded` should achieve the same purpose, I'm open to discussion on the implications. I leaned on `failed` because the type of error that is caught on this part is generally [not recoverable](https://github.com/elastic/beats/blob/488c445ed746fcaa6543ab52acb4a77ef3d39dce/heartbeat/monitors/monitor.go#L165).

Reply:
I was worried if it could stop the other monitors. But if thats not the case, I am not super inclined towards changing this.